server/src/tenant/entity-resolver/entity-resolver.service.ts

```diff
@@ -93,18 +101,178 @@ export class EntityResolverService {
     const graphqlResult = await workspaceDataSource?.query(`
       SELECT graphql.resolve($$
         {
-          ${entityName}Collection: : ${tableName}Collection(filter: { id: { eq: "${args.id}" } }) {
+          findOne${pascalCase(
+            entityName,
+          )}: ${tableName}Collection(filter: { id: { eq: "${args.id}" } }) {
             ${graphqlQuery}
           }
         }
       $$);
     `);
 
     const result =
-      graphqlResult?.[0]?.resolve?.data?.[`${entityName}Collection`];
+      graphqlResult?.[0]?.resolve?.data?.[`findOne${pascalCase(entityName)}`];
+
+    if (!result) {
+      throw new BadRequestException('Malformed result from GraphQL query');
+    }
+
+    return result;
+  }
+
+  async createOne(
+    entityName: string,
+    tableName: string,
+    args: { data: any },
+    workspaceId: string,
+    info: GraphQLResolveInfo,
+    fieldAliases: Record<string, string>,
+  ) {
+    if (!this.environmentService.isFlexibleBackendEnabled()) {
+      throw new ForbiddenException();
+    }
+
+    const workspaceDataSource =
+      await this.dataSourceService.connectToWorkspaceDataSource(workspaceId);
+
+    const graphqlQuery = await this.prepareGrapQLQuery(
+      workspaceId,
+      info,
+      fieldAliases,
+    );
+
+    await workspaceDataSource?.query(`
+      SET search_path TO ${this.dataSourceService.getSchemaName(workspaceId)};
+    `);
+    const graphqlResult = await workspaceDataSource?.query(`
+      SELECT graphql.resolve($$
+          mutation {
+            createOne${pascalCase(
+              entityName,
+            )}: insertInto${tableName}Collection(objects: [${stringify({
+      id: uuidv4(),
+      ...args.data,
+    })}]) {
+              affectedCount
+              records {
+                ${graphqlQuery}
+              }
+            }
+          }
+      $$);
+    `);
+
+    const result =
+      graphqlResult?.[0]?.resolve?.data?.[`createOne${pascalCase(entityName)}`]
+        ?.records[0];
 
     if (!result) {
-      return null;
+      throw new BadRequestException('Malformed result from GraphQL query');
+    }
+
+    return result;
+  }
+
+  async createMany(
```

Comment:
Nit: I feel like createOne can call createMany and remove duplicated code making it easier to maintain? Wdyt?

Reply:
@Weiko Yes, you're right, I separate them at first because argument generation was not the same at the beginning